### ISSUE TITLE
[Bug]: Error after updating to `v2.12.0`

### ISSUE THREAD

**Issue by Aumnescio** [mrjones2014/legendary.nvim#395]
### Similar Issues

- [X] Before filing, I have searched for similar issues.

### Neovim Version

```
NVIM v0.9.0
Build type: Release
LuaJIT 2.1.0-beta3
```

### Steps to Reproduce

1. Update to version 2.12.0
2. Bind:
```
{ "<C-p>",  "<Cmd>Legendary<CR>", description = "not relevant", mode = "n", opts = { noremap = true } },
```
using Legendary.
3. Try to activate the binding.
4. It errors:
```
Error executing Lua callback: ...share/nvim/lazy/legendary.nvim/lua/legendary/ui/init.lua:34: attempt to index local 'opts'
 (a nil value)
stack traceback:
        ...share/nvim/lazy/legendary.nvim/lua/legendary/ui/init.lua:34: in function 'select_inner'
        ...share/nvim/lazy/legendary.nvim/lua/legendary/ui/init.lua:98: in function 'select'
        ...al/share/nvim/lazy/legendary.nvim/lua/legendary/init.lua:124: in function 'find'
        ...hare/nvim/lazy/legendary.nvim/lua/legendary/api/cmds.lua:38: in function <...hare/nvim/lazy/legendary.nvim/lua/l
egendary/api/cmds.lua:10>
```

See minimal init.lua below.

### Expected Behavior

No error. Things should work as they worked in the previous version.

### Actual Behavior

The binding does not work, and instead gives an error.

### Minimal Configuration to Reproduce

```lua
-- Place this file into:        `~/.config/nvim_repro/init.lua`

-- Then launch Neovim using:    `NVIM_APPNAME=nvim_repro nvim`
-- NOTE: I think this might require nvim version 0.9.0 or above.

-- NOTE: When using `NVIM_APPNAME=nvim_repro` to launch into this alternate Neovim config,
-- this data path will not conflict with or pollute the primary data folders.
local lazypath = vim.fn.stdpath("data") .. "/lazy/lazy.nvim"

-- Lazy Bootstrap:
if not vim.loop.fs_stat(lazypath) then
    vim.fn.system({
        "git",
        "clone",
        "--filter=blob:none",
        "https://github.com/folke/lazy.nvim.git",
        "--branch=stable", -- latest stable release
        lazypath,
    })
end

vim.opt.rtp:prepend(lazypath)

-- `lazy.nvim` config options
local lazy_opts = {
    root = vim.fn.stdpath("data") .. "/lazy",               -- Directory where plugins will be installed.
}

local plugins = {
    "folke/tokyonight.nvim",
    {
        "mrjones2014/legendary.nvim",
        version = "2.12.0",     -- This does not work.
        -- version = "2.11.0",  -- This works
        dependencies = {
            "nvim-telescope/telescope.nvim",
            "stevearc/dressing.nvim",
        },
    },
    {
        "nvim-telescope/telescope.nvim",
        dependencies = {
            "nvim-lua/plenary.nvim"
        },
    },
}

-- Lazy Setup:
require("lazy").setup(plugins, lazy_opts)

require("dressing").setup()

require("legendary").setup()

-- Bind this thing.
require("legendary").keymaps({
    {
        "<C-p>", "<Cmd>Legendary<CR>",
        description = "Legendary: Search/Find Keymaps/Commands/Autocommands",
        mode = "n", opts = { noremap = true }
    },
})
-- Try activate the binding. It errors on v2.12.0.

-- Colors
vim.opt.termguicolors = true
vim.cmd([[colorscheme tokyonight]])

-- End of File
```

My template is a bit different, but this should work very easily with the instructions at the top of the file. Was easier for me to do this way.

### Output from :LegendaryLog

<details>
<summary><code>:LegendaryLog</code> output (click to expand)</summary>

```
[Sat 08 Jul 2023 01:26:45 AM EEST][legendary.nvim] Launching select UI
[Sat 08 Jul 2023 01:26:02 AM EEST][legendary.nvim] Launching select UI
[Sat 08 Jul 2023 01:22:42 AM EEST][legendary.nvim] Launching select UI
[Sat 08 Jul 2023 01:20:43 AM EEST][legendary.nvim] Launching select UI
[Sat 08 Jul 2023 01:20:26 AM EEST][legendary.nvim] Launching select UI
[Sat 08 Jul 2023 01:20:08 AM EEST][legendary.nvim] Launching select UI
[Sat 08 Jul 2023 01:19:41 AM EEST][legendary.nvim] setup() parsed and applied all configuration.
[Sat 08 Jul 2023 01:18:20 AM EEST][legendary.nvim] Launching select UI
[Sat 08 Jul 2023 01:18:19 AM EEST][legendary.nvim] setup() parsed and applied all configuration.
[Sat 08 Jul 2023 01:18:10 AM EEST][legendary.nvim] setup() parsed and applied all configuration.
...
```
</details>

When the binding causes the error, only a "Launching select UI" entry is added to the log.

### Additional Details and/or Screenshots

Edit: The `require('legendary').find()` command also errors in the same manner.

**Comment by mrjones2014** [comment]
Sorry about that folks. I will get it fixed later today.

**Comment by mrjones2014** [comment]
Apologies once again, stupid mistake on my part. Fixed in [v2.12.1](https://github.com/mrjones2014/legendary.nvim/releases/tag/v2.12.1)
